lib/raster/raster_metadata.c

```diff
@@ -81,6 +82,86 @@ void Rast_write_vdatum(const char *name, const char *str)
     misc_write_line("vertical_datum", name, str);
 }
 
+/*!
+ * \brief Get a raster map's band reference metadata string
+ *
+ * Read the raster's band reference metadata file and put string in str
+ *
+ * \param name raster map name
+ * \param mapset mapset name
+ *
+ * \return  string representing band reference on success
+ * \return  NULL on error
+ */
+char *Rast_read_bandref(const char *name, const char *mapset)
+{
+    return misc_read_line("bandref", name, mapset);
+}
+
+/*!
+ * \brief Write a string into a raster's band reference metadata file
+ *
+ * Raster map must exist in the current mapset.
+ *
+ * It is up to the caller to validate band reference string in advance
+ * with Rast_legal_bandref().
+ *
+ * \param name raster map name
+ * \param str  string containing data to be written
+ */
+void Rast_write_bandref(const char *name, const char *str)
+{
+    misc_write_line("bandref", name, str);
+}
+
+/*!
+ * \brief Check for legal band reference
+ *
+ * Legal band identifiers must be legal GRASS file names.
+ * They are in format <shortcut>_<bandname>.
+ * Band identifiers are capped in legth to GNAME_MAX.
+ *
+ * This function will return -1 if provided band id is not considered
+ * to be valid.
+ * This function does not check if band id maps to any entry in band
+ * metadata files as not all band id's have files with extra metadata.
+ *
+ * The function prints a warning on error.
+ *
+ * \param bandref band reference to check
+ *
+ * \return 1 success
+ * \return -1 failure
+ */
+int Rast_legal_bandref(const char *bandref)
+{
+    char **tokens;
+    int ntok;
+
+    if (strlen(bandref) >= GNAME_MAX - 1) {
```

Comment:
`strlen(bandref) >= GNAME_MAX` should be more correct.

`GNAME_MAX=256` which *includes* null string (i.e. 255 + 1), but strlen() returns length *without* the terminating character. If strlen() results in 256 or more the file name is too long.

Reply:
Can't remember my rationale behind the "-1" + no notes => adjusted to your recommendation.